comp/logs/agent/config/config_keys.go

```diff
@@ -121,26 +120,34 @@ func (l *LogsConfigKeys) hasAdditionalEndpoints() bool {
 	return len(l.getAdditionalEndpoints()) > 0
 }
 
-// getLogsAPIKey provides the dd api key used by the main logs agent sender.
-func (l *LogsConfigKeys) getLogsAPIKey() string {
-	if configKey := l.getConfigKey("api_key"); l.isSetAndNotEmpty(configKey) {
-		return pkgconfigutils.SanitizeAPIKey(l.getConfig().GetString(configKey))
+// getAPIKeyGetter returns a getter function to retrieve the API key from the configuration. The getter will refetch the
+// value from the configuration upon each call to ensure the latest version is used. This ensure that the logs agent is
+// compatible with rotating the API key at runtime.
+//
+// The getter will use "logs_config.api_key" over "api_key" when needed.
+func (l *LogsConfigKeys) getAPIKeyGetter() func() string {
```

Comment:
So the need for this is due to the LogsAgent using a read-only Config, right? It's unfortunate that we can't use the same notification based system, but this is good enough for getting the proper functionality.

I think this code would be better if it used some abstraction or interface, rather than returning a function, but that can be done as a follow-up. I'm thinking something like this:

```
type LiveString struct {
  resolve func() string
}

func (l *LiveString) GetCurrent() string {
  return l.resolve()
}

func NewLiveString(f func() string) *LiveString ...
func NewLiveStringFromString(s string) *LiveString ...
```

With Endpoint having a `APIKey LiveString` rather than `apiKeyGetter func() string`